src/extension/editorMessages.ts

```diff
@@ -31,6 +31,9 @@ export namespace EditorMessages {
     export const reportUnexpectedError = (errorDescription: string) =>
         `Coqpilot got an unexpected error: ${errorDescription}. Please report this crash by opening an issue in the Coqpilot GitHub repository.`;
 
+    export const completionAborted = 
+        "Completion generation was forcefully aborted, Coq-LSP server stopping. Please hit the Status Bar button again to restart the server and proceed with completions.";
```

Comment:
I just wonder, whether the Coq-LSP server could be restarted if the user misses / closes this message?

Reply:
And what do you mean about hitting the "Status Bar"? Isn't it the circle that spins during generation? 
Finally, I didn't find the code that restarts the server, how does the recovery work?

Reply:
It was implemented in another branch and therefore the implementation here was only partial( Sorry for misguiding you. I would merge this PR and update the Code in core-lsp-refactor branch and everything will get clear